### PR TITLE
Enable coverage testing for TS provider

### PR DESCRIPTION
**Pull request by ionut-arm**
This commit enables collection of code coverage results for the TS
provider, since current changes in the way testing is done means there
is no service restarting, which was triggering the problems before.
The nightly run manual toggle also has an added field, allowing the
tests to be run against specific revision numbers.

Signed-off-by: Ionut Mihalcea <ionut.mihalcea@arm.com>

### CODE REVIEW

File: .github/workflows/nightly.yml
Comment (hug-dev) [inline .github/workflows/nightly.yml:12]: Is it to manually select a specific commit to test?

**Review by hug-dev — APPROVED** [review]
😎

File: .github/workflows/nightly.yml
Comment (ionut-arm) [reply .github/workflows/nightly.yml:12]: Indeed - because if I want to manually test this change, for example, I have to wait until it gets merged.... not great.

**Pull request merged**
